### PR TITLE
10-Add offer dialog

### PR DESCRIPTION
**Pull request by DonutPirate**
Added note icon and seat picker to offer dialog

### CODE REVIEW

**Comment by Natoman95** [comment]
Right now I'm able to type in negative numbers on the dialog. What happened to the little indexing buttons?

Are you going to add the note avatar icon to the other dialog?

**Comment by DonutPirate** [comment]
The indexing buttons are on the far right of the input field. You can manually enter an invalid value but with those buttons it functions properly. Our two options are to allow for the users to type in any number here and throw the error if it's >1, or to have a drop-down menu with a limited range of options (and arbitrary max passenger amount) instead.

**Comment by Natoman95** [comment]
I wonder if there's a way that we can add the up and down arrows and prevent users from typing what they want. That way you could limit the inputs to a certain exact range without the messiness of a dropdown.

**Comment by Natoman95** [comment]
I'm not sure the up and down buttons are visible when you shrink it down to phone size. They may also just be way too small to click with a thumb quite honestly.

**Comment by DonutPirate** [comment]
I'm of the opinion that a dropdown is the best solution in every way, except for the fact that we have to cap it at a certain range which is hard to decide on since different vehicles have different capacities. Maybe we can cut it off at 10 since some of the largest SUVs have up to 9 seats.

**Comment by Natoman95** [comment]
Yeah, 10 sounds reasonable. I'm not opposed to a dropdown. You could also do something like this where the number itself can only be changed by buttons:

demo

**Comment by DonutPirate** [comment]
Ooh I like that better. You have a link or name of that element?

**Comment by Natoman95** [comment]
That’s not done via material ui, but you could probably find a way to do it in material ui. You could put two buttons on either side of a number that is indexed whenever you hit the buttons. Use a few "onClick()" methods and have a few if statements inside those so they never exceed a certain maximum or minimum.

**Comment by Natoman95** [comment]
I bet you could even get the buttons to sit right up next to the number box by setting the "paddingLeft: 0" and "paddingRight: 0" respectively on the two buttons and the number box. You may have to do the same with the margins as well.

**Comment by Natoman95** [comment]
@gordon-cs/cps492-2018 feel free to join in. We're discussing the best way to create a number picker that won't allow negative or non-number inputs that are less than a certain quantity. The interface should be easy to use on a mobile device.

**Comment by mFelgate** [comment]
Oh I can do that.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by Natoman95** [comment]
@mFelgate was telling me that there's probably a package already out there for react that does this. For example: https://www.npmjs.com/package/semantic-ui-react-numberpicker. All you'd need to do is install that NPM package, run yarn install, and then use this code that's already been built. If we ever need a visual component not available in material ui, the NPM site is the first place we should look.

**Comment by SamuelColacchia** [comment]
That library looks cool, though it is lacking api documentation.

**Comment by DonutPirate** [comment]
Got it working with a constant and some new functions. Didn't need to import a library.

**Comment by Natoman95** [comment]
That's even better. If it can be done elegantly without an external library that's great. Looks excellent. But could we refactor the javascript handlers to not be entirely in-line? Conditionals are really difficult to decipher when they're all on one line.

**Comment by DonutPirate** [comment]
Alright, ready to approve unless you have other suggestions.

File: goco-transit/src/components/offers.js [hunk 1]
```diff
@@ -54,6 +63,16 @@ class Offers extends React.Component {
     this.setState({ deleteOpen: false });
   };
 
+  // Limits seat maximum to pre-defined constant
+  handleSeatPlus = () => {
+    if (this.state.seats < 10) { this.setState({ seats: this.state.seats + 1 }) }
```
Comment (Natoman95) [inline goco-transit/src/components/offers.js]: This is better. I still prefer to split if-clauses onto multiple lines, but at least this in-line one is readable.

Comment (DonutPirate) [reply goco-transit/src/components/offers.js]: I split them up if it's more than one statement, but I don't mind either way for one-liners. We can go through towards the end and standardize our syntax if we want.

**Comment by Natoman95** [comment]
I think it's good to go, but I'll leave this open in case other people have suggestions. I've attached a picture of the dialog since people probably don't want to have to download our source code and figure out how to run the app themselves.

screen shot 2018-02-09 at 1 37 03 pm

**Pull request merged**